From 12c369f1084fd40542268e5bbf6f7d55bd242033 Mon Sep 17 00:00:00 2001
From: Yangbo Lu <yangbo.lu@nxp.com>
Date: Fri, 18 Oct 2019 18:17:45 +0800
Subject: [PATCH 2/2] Revert "lib/boilerplate: Prevent migration to secondary
 mode for __printout"

This reverts commit 9da7bde25afaf5d6fb7f91a44bc2e96019cd4acb.
---
 lib/boilerplate/ancillaries.c | 29 ++++++-----------------------
 1 file changed, 6 insertions(+), 23 deletions(-)

diff --git a/lib/boilerplate/ancillaries.c b/lib/boilerplate/ancillaries.c
index 7e98141..13d7faf 100644
--- a/lib/boilerplate/ancillaries.c
+++ b/lib/boilerplate/ancillaries.c
@@ -52,30 +52,14 @@ static void __do_printout(const char *name, const char *header,
 {
 	FILE *fp = stderr;
 
-	__RT(fprintf(fp, "%4u\"%.3u.%.3u| ", ms / 1000, ms % 1000, us));
+	fprintf(fp, "%4u\"%.3u.%.3u| ", ms / 1000, ms % 1000, us);
 
 	if (header)
-		__RT(fputs(header, fp));
+		fputs(header, fp);
 
-	__RT(fprintf(fp, "[%s] ", name ?: "main"));
-	__RT(vfprintf(fp, fmt, ap));
-	__RT(fputc('\n', fp));
-}
-
-static void __do_early_printout(const char *name, const char *header,
-			  unsigned int ms, unsigned int us,
-			  const char *fmt, va_list ap)
-{
-	FILE *fp = stderr;
-
-	__STD(fprintf(fp, "%4u\"%.3u.%.3u| ", ms / 1000, ms % 1000, us));
-
-	if (header)
-		__STD(fputs(header, fp));
-
-	__STD(fprintf(fp, "[%s] ", name ?: "main"));
-	__STD(vfprintf(fp, fmt, ap));
-	__STD(fputc('\n', fp));
+	fprintf(fp, "[%s] ", name ?: "main");
+	vfprintf(fp, fmt, ap);
+	fputc('\n', fp);
 	fflush(fp);
 }
 
@@ -93,7 +77,7 @@ void __printout(const char *name, const char *header,
 	 * uncontended.
 	 */
 	if (!init_done) {
-		__do_early_printout(name, header, 0, 0, fmt, ap);
+		__do_printout(name, header, 0, 0, fmt, ap);
 		return;
 	}
 	
@@ -126,7 +110,6 @@ void ___panic(const char *fn, const char *name,
 	if (asprintf(&p, "BUG in %s(): ", fn) < 0)
 		p = "BUG: ";
 	__printout(name, p, fmt, ap);
-	rt_print_flush_buffers();
 	exit(1);
 }
 
-- 
2.7.4

